src/shims/unix/unnamed_socket.rs

```diff
@@ -68,8 +68,10 @@ impl FileDescription for AnonSocket {
 
         // Check if it is readable.
         if let Some(readbuf) = &self.readbuf {
-            if !readbuf.borrow().buf.is_empty() {
+            let buffer = readbuf.borrow();
+            if !buffer.buf.is_empty() {
                 epoll_ready_events.epollin = true;
+                epoll_ready_events.ready_events_clock.join(&buffer.clock)
```

Comment:
Just curious, why ``join`` is used here instead of ``release_clock``?

Reply:
I believe because the clock is being merged. There is no happens-before at this stage. Another way to view it is neither of the two objects is a thread. Just the glue to get the clock from the source thread to the destination thread.